tools/HiveConsensusWorkflowGenerator/Program.cs

```diff
@@ -1,45 +1,91 @@
+using System.Diagnostics;
+using System.Text.Json;
+
 namespace HiveConsensusWorkflowGenerator;
 
 public static class Program
 {
-    const long MaxSizeWithoutSplitting = 35_000_000;
-    const long TargetSize = 23_000_000;
-    const long PenaltyForAdditionalInit = 300_000;
+    const int MaxJobsCount = 256;
 
     static void Main(string[] args)
     {
         string path = args.FirstOrDefault() is not null ? args.First() : "src/tests";
 
         IEnumerable<string> directories = GetTestsDirectories(path);
         Dictionary<string, long> pathsToBeTested = GetPathsToBeTested(directories);
-        IEnumerable<List<string>> accumulatedJobs = GetTestsSplittedToJobs(pathsToBeTested);
 
-        TextWriter fileContent = CreateTextWriter();
+        // Sort the tests by size in descending order
+        var sortedTests = pathsToBeTested.OrderByDescending(kv => kv.Value).ToList();
+
+        // Create empty list of 256 jobs
+        var groupedTestNames = new List<List<(string fullPath, string fileName)>>(); // Store both full path and filename
+        for (int i = 0; i < MaxJobsCount; i++)
+        {
+            groupedTestNames.Add(new List<(string, string)>());
+        }
 
-        WriteInitialLines(fileContent);
+        File.WriteAllText("allTests.json", string.Join('\n', sortedTests.Select(x => x.Key)));
+
+        // Initial distribution without size limit
+        int groupIndex = 0;
+        foreach (var test in sortedTests.Take(MaxJobsCount))
+        {
+            string fileName = test.Key;
+            groupedTestNames[groupIndex].Add((test.Key, fileName));
+            groupIndex = (groupIndex + 1) % MaxJobsCount;
+        }
 
-        int jobsCreated = 0;
-        foreach (List<string> job in accumulatedJobs)
+        // Subsequent distribution with size limit
+        foreach (var test in sortedTests.Skip(MaxJobsCount))
         {
-            WriteJob(fileContent, job, ++jobsCreated);
+            string fileName = test.Key;
+            groupIndex = FindSuitableGroupIndex(groupedTestNames, pathsToBeTested, test.Value);
```

Comment:
Minor: `groupedTestNames` could be reworked to be sorted collection so that the smallest one would always be the first one.

Reply:
@LukaszRozmej 
Applied SortedList - pretty nice, simplified implementation a bit but needed to use one hack for that (to make sure Key is unique as it is long with fizeSizes)
Please take a look, I guess i missed something but overall seems like implementation works just fine.